src/node.js

```diff
@@ -386,6 +404,51 @@
       } while (tickInfo[kLength] !== 0);
     }
 
+    function doNTCallback0(callback) {
+      var threw = true;
+      try {
+        callback();
+        threw = false;
+      } finally {}
+      return threw;
```

Comment:
in `_tickCallback` there's [code in the finalizer](https://github.com/iojs/io.js/blob/master/src/node.js#L344-L347) – if control passes to the finalizer, `tickDone` is called before the exception propagates as usual. It should be replicable by moving the `try {} finally { if (threw) tickDone() }` back up into `_tickCallback`, wrapping [these lines](https://github.com/mscdex/io.js/blob/fa69a24ec5f5c72e581a3069989c38c4ace4cf06/src/node.js#L338-L354) in the TryClause.